src/layer/tile/TileLayer.js

```diff
@@ -92,6 +92,18 @@ export const TileLayer = GridLayer.extend({
 
 		options = Util.setOptions(this, options);
 
+		// in case the attribution hasn't been specified, check for known hosts that require attribution
+		if (options.attribution === null && URL.canParse(url)) {
+			const urlHostname = new URL(url).hostname;
+
+			// check for Open Street Map hosts
+			const osmHosts = ['tile.openstreetmap.org', 'tile.osm.org'];
+			const isOsmUrl = osmHosts.some(host => urlHostname.endsWith(host));

```

Comment:
I'd rather get rid of `const isOsmUrl`, and do `if (osmHosts.some(...)) {` instead.

Reply:
I think the `const isOsmUrl` makes the code slightly more readable, describing what the `osmHosts.some` accomplishes. But I have no strong opinions, I can change it if you think inline it's better.

Reply:
I see it like you @Jouwee , it makes it more readable. But lets make the change, then I will merge it

Reply:
Change is made. I also see that a test has failed, but it seems to be a flaky test?